app/pb_kits/playbook/pb_user/user.rb

```diff
@@ -0,0 +1,117 @@
+                 configured_id
+                 configured_name
+                 configured_orientation
+                 configured_size
+                 configured_title].freeze
+
+      def initialize(align: default_configuration,
+                     avatar: default_configuration,
+                     classname: default_configuration,
+                     data: default_configuration,
+                     id: default_configuration,
+                     name: default_configuration,
+                     orientation: default_configuration,
+                     size: default_configuration,
+                     title: default_configuration)
+        self.configured_align = align
+        self.configured_avatar = avatar
+        self.configured_classname = classname
+        self.configured_data = data
+        self.configured_id = id
+        self.configured_name = name
+        self.configured_orientation = orientation
+        self.configured_size = size
+        self.configured_title = title
+      end
+
+      def align
+        align_options = %w[left center right]
+        one_of_value(configured_align, align_options, "left")
+      end
+
+      def avatar_size
+        case size
+        when "lg"
+          "xl"
+        when "md"
+          "md"
+        else
+          "sm"
+        end
+      end
+
+      def avatar
+        if is_set? configured_avatar
+          avatar_obj = configured_avatar == true ? {} : configured_avatar
+          avatar_obj[:size] = avatar_size
+          avatar_obj[:name] = configured_name if is_set? configured_name
+          pb_avatar = Playbook::PbAvatar::Avatar.new(avatar_obj)
+          ApplicationController.renderer.render(partial: pb_avatar, as: :object)
+        end
+      end
+
+      def name_size
+        size == "lg" ? 3 : 4
+      end
+
+      def name
+        if is_set? configured_name
+          title_props = { text: configured_name, size: name_size }
+          pb_name = Playbook::PbTitle::Title.new(title_props)
+          ApplicationController.renderer.render(partial: pb_name, as: :object)
+        end
+      end
+
+      def orientation
+        orientation_options = %w[vertical horizontal]
+        one_of_value(configured_orientation, orientation_options, "horizontal")
+      end
+
+      def size
+        size_options = %w[lg md sm]
+        one_of_value(configured_size, size_options, "sm")
+      end
+
+      def title
+        if is_set? configured_title
+          pb_icon_element = Playbook::PbBody::Body.new(color: "light") do
+            configured_title
+          end
+          ApplicationController.renderer.render(partial: pb_icon_element, as: :object)
+        end
+      end
+
+      def kit_class
+        kit_options = [
+          "pb_user",
+          align,
+          orientation,
+          size,
+        ]
+        kit_options.join("_")
+      end
+
+      def to_partial_path
+        "pb_user/user"
+      end
+
+    private
+
+      DEFAULT = Object.new
```

Comment:
Would it be enough to have `default_configuration` to just return `Object.new`?

Reply:
@drborges This pattern works because `Object.new != Object.new`. The value of Object.new has to be captured so it can be compared to other values.

Since `initialize` is an instance method, we could rewrite this using a private instance method that captures the value of `Object.new` into a instance variable.

That would look something like:
```ruby
class Example
  def initialize(example_option: default)
    ...
  end

  private

  def default
    @default ||= default
  end
end
```

The only real downside to this, is that we will be creating more instances of `Object.new`, but that is really not a big deal, and I would not be concerned with it.

Reply:
Yup, that's what I had in mind. I'm just wondering if the existing boilerplate is worth the trouble...

Reply:
I think I am picking up what your putting down, and too am questing if the current version is worth the boiler plate.